### PR TITLE
cleanup: Use target_link_libraries directly in cmake.

### PR DESCRIPTION
**Pull request by iphydf**
Instead of using target_link_modules, which does magic that we no longer need, because we only have 1 library we install, and all binaries we build link statically because they need access to internal symbols.

- - -
This change is [Reviewable](https://reviewable.io/reviews/TokTok/c-toxcore/2499)

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -422,8 +427,8 @@ install_module(toxcore DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/tox)
 
 function(unit_test subdir target)
   add_executable(unit_${target}_test ${subdir}/${target}_test.cc)
-  target_link_modules(unit_${target}_test toxcore)
-  target_link_libraries(unit_${target}_test GTest::GTest GTest::Main)
+  target_link_libraries(unit_${target}_test toxcore_static)
```
Comment (Green-Sky) [inline CMakeLists.txt:430]: missing visibility (`PRIVATE`)

File: cmake/ModulePackage.cmake [hunk 2]
```diff
@@ -62,10 +62,14 @@ function(add_module lib)
   if(ENABLE_SHARED)
     add_library(${lib}_shared SHARED ${ARGN})
     set_target_properties(${lib}_shared PROPERTIES OUTPUT_NAME ${lib})
+    add_library(${lib} ALIAS ${lib}_shared)
```
Comment (Green-Sky) [inline cmake/ModulePackage.cmake:65]: can we have namespaced aliases?
`ToxCore::${lib}`

see https://cmake.org/cmake/help/v3.11/manual/cmake-buildsystem.7.html#alias-targets

should make it easier to conditionally fetchconent/submodule toxcore in other projects

Comment (iphydf) [reply cmake/ModulePackage.cmake]: Yes, but not in this PR.

File: CMakeLists.txt (hunk 1 above)
Comment (iphydf) [reply CMakeLists.txt]: Done.

**Review by Green-Sky — APPROVED** [review]

**Pull request merged**